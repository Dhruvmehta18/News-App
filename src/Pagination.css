.pagination:nth-child(n):hover:not(.active){
  background-color: #ddd;
  cursor: pointer;
}

.pagination:nth-child(n):active{
  color: white;
  background-color: #3f51b5;
  cursor: pointer;
}
.active{
  color: white;
  background-color: #3f51b5;
  cursor: pointer;
}
