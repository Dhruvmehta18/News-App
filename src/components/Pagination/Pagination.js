import React, { Component } from "react";
import Box from "@material-ui/core/Box";
import "./Pagination.css"
class Pagination extends Component {
  render(){
    const {
      pageNo,
      nbPages,
      OnPaginationItemClickListener,
    }=this.props;
  let pageArray=[];
  const range=4;
  let temp=(pageNo-range);
  const min=temp>1?temp:2;
  temp=min+2*range;
  const max=temp<50?(nbPages<2*range?nbPages:temp):49;
  // console.log("min="+min+" max="+max);
  for (let index = min; index <= max; index++) {
    pageArray[index]=index;
  }
  return(
    <div className="interactions">
    <Box display="flex" p={1} m={1} justifyContent="center" >
      {pageNo===0&&
        <Box p={1} key={0} className="pagination active" onClick={() => OnPaginationItemClickListener(0)} >
        1
      </Box>}
      {pageNo!==0&&
      <Box p={1} key={0} className="pagination" onClick={() => OnPaginationItemClickListener(0)} >
        1
      </Box>}
      {min!==2&&
        <Box p={1}  key={-1} className="pagination">
      &lsaquo;&lsaquo;
      </Box>}
      
      {pageArray.map((item,index)=>{
        const className = pageNo === index-1 ? 'pagination active' : 'pagination';
        return(
        <Box p={1}  key={item-1} className={className} onClick={() => OnPaginationItemClickListener(item-1)}>
        {item}
        </Box>
        )
      })
      }
      {max!==49&&max!==min-1&&
        <Box p={1} key={nbPages} className="pagination" >
        &rsaquo;&rsaquo;
        </Box>}
      {max!==min-1&&pageNo!==nbPages-1&&
      <Box p={1}  key={nbPages-1} className="pagination" onClick={() => OnPaginationItemClickListener(nbPages-1)}>
        {nbPages}
      </Box>
    }
    {max!==min-1&&pageNo===nbPages-1&&
      <Box p={1}  key={nbPages-1} className="pagination active" onClick={() => OnPaginationItemClickListener(nbPages-1)}>
        {nbPages}
      </Box>
    }
    </Box>
    </div>
  );
}
}
export default Pagination;