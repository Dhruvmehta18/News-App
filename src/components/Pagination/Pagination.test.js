import React from 'react';
import ReactDOM from 'react-dom';
import Pagination from './Pagination'
import renderer from 'react-test-renderer';

describe('Pagination', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Pagination/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  test('has a valid snapshot', () => {
      const pageNo = 1,nbPages=50;
    const component = renderer.create(
      <Pagination pageNo={pageNo} nbPages={nbPages} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
