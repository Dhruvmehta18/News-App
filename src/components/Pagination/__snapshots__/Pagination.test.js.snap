// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination has a valid snapshot 1`] = `
<div
  className="interactions"
>
  <div
    className="MuiBox-root MuiBox-root-16"
  >
    <div
      className="MuiBox-root MuiBox-root-17 pagination"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="MuiBox-root MuiBox-root-18 pagination active"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="MuiBox-root MuiBox-root-19 pagination"
      onClick={[Function]}
    >
      3
    </div>
    <div
      className="MuiBox-root MuiBox-root-20 pagination"
      onClick={[Function]}
    >
      4
    </div>
    <div
      className="MuiBox-root MuiBox-root-21 pagination"
      onClick={[Function]}
    >
      5
    </div>
    <div
      className="MuiBox-root MuiBox-root-22 pagination"
      onClick={[Function]}
    >
      6
    </div>
    <div
      className="MuiBox-root MuiBox-root-23 pagination"
      onClick={[Function]}
    >
      7
    </div>
    <div
      className="MuiBox-root MuiBox-root-24 pagination"
      onClick={[Function]}
    >
      8
    </div>
    <div
      className="MuiBox-root MuiBox-root-25 pagination"
      onClick={[Function]}
    >
      9
    </div>
    <div
      className="MuiBox-root MuiBox-root-26 pagination"
      onClick={[Function]}
    >
      10
    </div>
    <div
      className="MuiBox-root MuiBox-root-27 pagination"
    >
      ››
    </div>
    <div
      className="MuiBox-root MuiBox-root-28 pagination"
      onClick={[Function]}
    >
      50
    </div>
  </div>
</div>
`;
